Mathlib/Data/Matrix/Basic.lean

```diff
@@ -1209,6 +1210,87 @@ def diagonalAlgHom : (n → α) →ₐ[R] Matrix n n α :=
 
 end Algebra
 
+section AddHom
+
+variable [Add α]
+
+variable (R α) in
+/-- Extracting entries from a matrix as an additive homomorphism.  -/
+@[simps]
+def elemAddHom (i : m) (j : n) : AddHom (Matrix m n α) α where
+  toFun M := M i j
+  map_add' _ _ := rfl
+
+-- The type ascription on the RHS is necessary for unification to succeed on the composition.
+lemma elemAddHom_eq_comp {i : m} {j : n} :
+    elemAddHom α i j = (Pi.evalAddHom _ j).comp (Pi.evalAddHom _ i : AddHom _ (n → α)) :=
```

Comment:
Arguably you should launder through `of` here to avoid needing the type ascription.

Reply:
I'm not quite sure I follow - `of` is for building matrices, but both sides here are maps out of a matrix space. I'm happy to use `of`, but I don't see how it fits here.

Reply:
You could add an `ofLinearEquiv` as a linear version of `of`.

Reply:
Sounds potentially helpful but a can of worms I'd rather not open here (AddEquiv; LinearEquiv; RingEquiv; AlgebraEquiv...). Would you be happy with a todo instead?

Reply:
I think it's fine to just add the bundling you need; the simp-normal form would remain `of`

Reply:
The add monoid hom one end up looking like this
```
lemma entryAddMonoidHom_eq_comp {i : m} {j : n} :
    entryAddMonoidHom α i j =
      ((Pi.evalAddMonoidHom _ j).comp (Pi.evalAddMonoidHom _ i)).comp
        (AddMonoidHomClass.toAddMonoidHom ofAddEquiv.symm) :=
  rfl
```
Are you sure this is a good idea?

Reply:
> You could add an `ofLinearEquiv` as a linear version of `of`.

I needed this anyway in #18097.

Reply:
> Are you sure this is a good idea?

Yes, because it means if you apply this lemma then use `simp`, the `of` reappears where it should.

Reply:
Okay, I added it in the most recent commit (from two weeks ago)